int32 height
int32 width
int32 size
int32 depth
int32 max_depth

QuadTreeNode[] nodes
